drivers/sensor/wsen_hids/wsen_hids.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2022 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_hids
+
+#include <string.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+#include "wsen_hids.h"
+
+LOG_MODULE_REGISTER(WSEN_HIDS, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * List of supported output data rates (sensor_value struct, input to
+ * sensor_attr_set()). Index into this list is used as argument for
+ * HIDS_setOutputDataRate().
+ */
+static const struct sensor_value hids_odr_list[] = {
+	{.val1 = 0, .val2 = 0},
+	{.val1 = 1, .val2 = 0},
+	{.val1 = 7, .val2 = 0},
+	{.val1 = 12, .val2 = 5 * 100000},
+};
+
+static int hids_sample_fetch(const struct device *dev, enum sensor_channel channel)
+{
+	struct hids_data *data = dev->data;
+	int16_t rawHumidity;
+	int16_t rawTemp;
+
+	__ASSERT_NO_MSG(channel == SENSOR_CHAN_ALL);
+
+	if (HIDS_getRawValues(&data->sensor_interface, &rawHumidity, &rawTemp) != WE_SUCCESS) {
+		LOG_ERR("Failed to fetch data sample.");
+		return -EIO;
+	}
+
+	if (HIDS_convertHumidity_uint16(&data->sensor_interface, rawHumidity, &data->humidity) !=
+	    WE_SUCCESS) {
+		LOG_ERR("Failed to convert humidity sample.");
+		return -EIO;
+	}
+
+	if (HIDS_convertTemperature_int16(&data->sensor_interface, rawTemp, &data->temperature) !=
+	    WE_SUCCESS) {
+		LOG_ERR("Failed to convert temperature sample.");
```

Comment:
```suggestion
		LOG_ERR("Failed to %s sample.", "convert temperature");
```

This will save a few bytes in ROM

Reply:
Done.